crates/oxc_linter/src/rules/eslint/id_length.rs

```diff
@@ -0,0 +1,573 @@
+    /// function bar(...x) { }
+    /// function baz([x]) { }
+    /// const [z] = arr;
+    /// const { prop: [i]} = {};
+    /// function qux({x}) { }
+    /// const { j } = {};
+    /// const { prop: a} = {};
+    /// ({ prop: obj.x } = {});
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// /*eslint id-length: "error"*/     // default is minimum 2-chars ({ "min": 2 })
+    ///
+    /// const num = 5;
+    /// function _f() { return 42; }
+    /// function _func() { return 42; }
+    /// obj.el = document.body;
+    /// const foo = function (evt) { /* do stuff */ };
+    /// try {
+    ///     dangerousStuff();
+    /// } catch (error) {
+    ///     // ignore as many do
+    /// }
+    /// const myObj = { apple: 1 };
+    /// (num) => { num * num };
+    /// function bar(num = 0) { }
+    /// class MyClass { }
+    /// class Foo { method() {} }
+    /// class Bar { #method() {} }
+    /// class Baz { field = 1 }
+    /// class Qux { #field = 1 }
+    /// function baz(...args) { }
+    /// function qux([longName]) { }
+    /// const { prop } = {};
+    /// const { prop: [name] } = {};
+    /// const [longName] = arr;
+    /// function foobar({ prop }) { }
+    /// function foobaz({ a: prop }) { }
+    /// const { a: property } = {};
+    /// ({ prop: obj.longName } = {});
+    /// const data = { "x": 1 };  // excused because of quotes
+    /// data["y"] = 3;  // excused because of calculated property access
+    /// ```
+    IdLength,
+    eslint,
+    nursery, // TODO: change category to `correctness`, `suspicious`, `pedantic`, `perf`, `restriction`, or `style`
+             // See <https://oxc.rs/docs/contribute/linter.html#rule-category> for details
+);
+
+impl Rule for IdLength {
+    fn from_configuration(value: Value) -> Self {
+        let object = value.get(0).and_then(Value::as_object);
+
+        Self {
+            exception_patterns: object
+                .and_then(|map| map.get("exceptionPatterns"))
+                .and_then(Value::as_array)
+                .unwrap_or(&vec![])
+                .iter()
+                .map(|val| Regex::new(val.as_str().unwrap()).unwrap())
+                .collect(),
+            exceptions: object
+                .and_then(|map| map.get("exceptions"))
+                .and_then(Value::as_array)
+                .unwrap_or(&vec![])
+                .iter()
+                .map(|val| val.as_str().unwrap().to_string())
+                .collect(),
+            max: object.and_then(|map| map.get("max")).and_then(Value::as_u64).unwrap_or(u64::MAX),
+            min: object.and_then(|map| map.get("min")).and_then(Value::as_u64).unwrap_or(2),
+            properties: object
+                .and_then(|map| map.get("properties"))
+                .and_then(Value::as_str)
+                .map(PropertyKind::from)
+                .unwrap_or_default(),
+        }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
```

Comment:
I left this method linear so that it would be easy to read. It could probably be refactored into some smaller functions.

Reply:
Broke the method up into 3 smaller methods based on the top level AST node kinds. Might still be able to be refactored more, but I'm not seeing anything myself.